### PR TITLE
Unconditionally polyfill String#startsWith(), broken on Android

### PR DESCRIPTION
**Pull request by cjb**
@keybase/react-hackers 

This was fun! We were redboxing on creating a new Chat sometimes on Android, and it ultimately came down to `conversationIDKey.startsWith('__pendingConversation__')` returning false when it ought to return true (because the string really did start with that).

So let's polyfill.  @chrisnojima, is this the best place to do a polyfill?  Want it to happen pretty early.

### CODE REVIEW

File: shared/index.native.js [hunk 1]
```diff
@@ -89,6 +89,13 @@ class Keybase extends Component {
 }
 
 function load () {
+  // Native String.startswith() sometimes incorrectly returns false on Android!
+  // $FlowIssue redefining startsWith
+  String.prototype.startsWith = function (searchString, position) { // eslint-disable-line no-extend-native
```
Comment (chrisnojima) [inline shared/index.native.js]: likely better to do it in globals.native. we import this on top of this file where we do some polyfilling already

Comment (chrisnojima) [inline shared/index.native.js]: also is this correct? maybe we should just use https://github.com/mathiasbynens/String.prototype.startsWith

Comment (cjb) [inline shared/index.native.js]: Thanks, moved it there and retested.

**Comment by cjb** [comment]
@chrisnojima I got this polyfill from MDN:

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/startsWith#Polyfill

The one that you linked does a bunch of stuff for IE compat down to ES3, didn't seem worth it.

**Pull request merged**